### ISSUE TITLE
Slack shouldn't stack populate calls?

### ISSUE THREAD

**Issue by KrzysztofMadejski** [42wim/matterbridge#760]
I'm not sure about that, but I think that Slack shouldn't stack user and channel populate calls.

in populateUsers if there is refresh in progress I believe the code shouldn't wait event if wait is set to true.. What is the use case for that? It may lead to stack of populate calls indefinitely if the `populate*` calls come more often than the populate gets to finish.

https://github.com/42wim/matterbridge/blob/57f156be8328529666aadaee780711f12b6e162b/bridge/slack/helpers.go#L102-L107

**Comment by 42wim** [comment]
@Helcaraxan is this still an issue after your recent changes?

**Comment by Helcaraxan** [comment]
Yes. It is unfortunately. There's no clear way of solving it. It'd rather put this down to be solved with my incoming proposal for unified user-management.
